### ISSUE TITLE
Create static cohorts

### ISSUE THREAD

**Issue by timgl** [PostHog/posthog#2627]
## Is your feature request related to a problem?

I want to be able to easily create cohorts from the list of users I got from doing a query.

## Describe the solution you'd like

Everywhere we can currently click on charts to see a list of users should have an option to create a static cohort.

- Insights
- Retention
- Funnels (an entire step)
- Lifecycle

**Comment by paolodamico** [comment]
Additional [revelant context](https://posthog.slack.com/archives/C0113360FFV/p1608291956004600). We tried doing this by using an API and some automation to create a cohort with 100+ match groups where each match group represented the exact match to the email address of the users we needed. This caused CH to crash.

The context on why we did this in the first place. We wanted to do retention analysis on a random sample of users to validate our hypothesis that certain user personas retain better than others. We did a random sample because for each user we had to know certain traits that require manual work. The random sample had to be done offline from a list of users obtained from a data export.
